File: app/src/main/legacy/io/dwak/holohackernews/app/dagger/component/AppComponent.java
package io.dwak.holohackernews.app.dagger.component;

import android.app.Application;
import android.content.res.Resources;

import javax.inject.Named;

import dagger.Component;
import io.dwak.holohackernews.app.cache.CacheManager;
import io.dwak.holohackernews.app.dagger.module.AppModule;

@Component(modules = {AppModule.class})
public interface AppComponent {
    void inject(Application application);

    @Named(AppModule.CACHE_MANAGER)
    CacheManager getCacheManager();

    @Named(AppModule.RESOURCES)
    Resources getResources();
}

File: app/src/mock/kotlin/io/dwak/holohackernews/app/dagger/component/AppComponent.kt
package io.dwak.holohackernews.app.dagger.component

import dagger.Component
import io.dwak.holohackernews.app.HackerNewsApplication
import io.dwak.holohackernews.app.dagger.module.AppModule
import io.dwak.holohackernews.app.dagger.module.MockService
import io.dwak.holohackernews.app.ui.list.presenter.StoryItemPresenterImpl
import timber.log.Timber

@Component(modules = arrayOf(AppModule::class))
interface AppComponent {
    fun inject(storyItemPresenter : StoryItemPresenterImpl)
    fun inject(mockService : MockService)

    val app : HackerNewsApplication

    val tree : Timber.Tree

}